lib/rubocop/cop/style/arguments_forwarding.rb

```diff
@@ -289,6 +286,27 @@ def classification
 
           private
 
+          def can_forward_all?
+            return false if any_arg_referenced?
+            return false if ruby_32_missing_rest_or_kwest?
+            return false unless offensive_block_forwarding?
+            return false if forward_additional_kwargs?
+
+            no_additional_args? || (target_ruby_version >= 3.0 && no_post_splat_args?)
+          end
```

Comment:
This is the meat of the change of the PR - with various additional conditions as to when a forward-all is or is not allowed